apiserver/facades/client/client/status.go

```diff
@@ -743,6 +743,15 @@ func fetchAllApplicationsAndUnits(
 		}
 	}
 
+	// Get all the space infos before iterating over the application infos.
```

Comment:
This is more of a patch comment as opposed to something I would leave in the code-base. The code alone is explanation enough.

This is a nice gain, but a further (micro) optimisation would be to acquire it even higher in the call stack and pass it to `fetchNetworkInterfaces` as well as this method.

Reply:
I did try this, but it does cause a lot of changes. I'll make a comment about that as a further optimization we can lift this up further.